src/lib.rs

```diff
@@ -322,6 +325,13 @@ pub fn count(i: usize) -> usize {
 /// assert_eq!(nodes, [7]);
 /// ```
 pub fn full_roots(i: usize, nodes: &mut Vec<usize>) {
+  assert!(
+    !((i & 1) > 0),
+    format!(
+      "You can only look up roots for depth 0 blocks, got index {}",
+      i
+    )
+  );
   if is_odd(i) {
     return;
   }
```

Comment:
Does `assert` always execute? If so, then `is_odd(i)` check can be removed.

Reply:
good one!